File: server/src/main/java/commons/Regex.java
package commons;

import java.util.regex.Pattern;

public abstract class Regex {
    public static Boolean isPhoneValid(String phone) {
        Pattern pattern = Pattern.compile("^(0|\\+33)[1-9]([-. ]?[0-9]{2}){4}$");
        return pattern.matcher(phone).find();
    }

    public static Boolean isMailValid(String mail) {
        Pattern pattern = Pattern.compile("^\\w+([.-]?\\w+)*@\\w+([.-]?\\w+)*(\\.\\w{2,3})+$");
        return pattern.matcher(mail).find();
    }

    public static Boolean isSiretValid(String siret) {
        Pattern pattern = Pattern.compile("^((RCS)?([0-9]{3}){3}$|^([0-9]{3}){3}[0-9]{5})|\\s*$");
        return pattern.matcher(siret).find();
    }

    public static Boolean isWebsiteUrlValid(String websiteUrl) {
        Pattern pattern = Pattern.compile("^((http://|https://)?(www.)?([a-zA-Z0-9]+).[a-zA-Z0-9]*.[a-z]{3}.?([a-z]+)?)|\\s*$");
        return pattern.matcher(websiteUrl).find();
    }

    public static Boolean isFrenchPostalCodeValid(String postalCode) {
        Pattern pattern = Pattern.compile("^([0-9]){5}$");
        return pattern.matcher(postalCode).find();
    }
}

File: client/src/commons/regex.js
export function isPhoneValid(phone) {
  return /^(?:(?:\+|00)33[\s.-]{0,3}(?:\(0\)[\s.-]{0,3})?|0)[1-9](?:(?:[\s.-]?\d{2}){4}|\d{2}(?:[\s.-]?\d{3}){2})$/.test(phone);
}

export function isMailValid(mail) {
  return /^\w+([.-]?\w+)*@\w+([.-]?\w+)*(\.\w{2,3})+$/.test(mail);
}

export function isSiretValid(siret) {
  return /^(RCS ?)?([0-9]{3} ?){2}[0-9]{3}$|^([0-9]{3} ?){3}[0-9]{5}$/.test(siret);
}

export function isWebsiteUrlValid(websiteUrl) {
  return /^((http(s)?:\/\/)[-a-zA-Z0-9:@;?&=\/%\+\.\*!'\(\),\$_\{\}\^~\[\]`#|]+)$/.test(websiteUrl);
}

export function isFrenchPostalCodeValid(postalCode) {
  return /^([0-9]){5}$/.test(postalCode);
}
